src/Service/RevisionSaver.php

```diff
@@ -36,6 +37,14 @@ public function __construct( MediawikiApi $api ) {
 	 * @returns bool success
 	 */
 	public function save( Revision $revision, EditInfo $editInfo = null ) {
+		if( is_null( $editInfo ) ) {
+			if( is_null( $revision->getEditInfo() )) {
+				$editInfo = new EditInfo();
+			} else {
+				$editInfo = $revision->getEditInfo();
+			}
+		}
```

Comment:
Since the second check is no longer necessary, it can be written as such. I will change it.